.github/workflows/codescan.yml

```diff
@@ -33,7 +33,7 @@ jobs:
           DOCKER_BUILDKIT=1 docker build -t docker.pkg.github.com/ossf/scorecard:${{ github.sha }} .
 
       - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@master
+        uses: aquasecurity/trivy-action@v0.0.15
```

Comment:
Instead of tag , can you please pin it to a sha? Tags can be moved

Reply:
Would love to but I'm note sure it's supported. 
In addition, the rest of the code uses versioning and not hash pin, so this is consistent and should be good enough for this PR. We can create a new PR to move all the code to using hash at once, if it's doable/supported.

Thoughts?

Reply:
> Would love to but I'm note sure it's supported.

Have you tried it? It is supported, I have used it in the past 

https://julienrenaux.fr/2019/12/20/github-actions-security-risk/#solution-use-commit-hash-as-version

> In addition, the rest of the code uses versioning and not hash pin, so this is consistent and should be good enough for this PR. We can create a new PR to move all the code to using hash at once, if it's doable/supported.
> 
> Thoughts?

I agree the rest of the actions code isn't using SHA, if we are fixing it then let's do it right.  Unless there are some issues in using the SHA.

Reply:
Yes, i didnt know either, this is definitely something we should do, scorcard should set the benchmark for secure configuration.
E.g. "actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4"
from https://dev.to/mheap/ensure-your-github-actions-are-pinned-to-a-sha-2d4k